src/Microsoft.AspNetCore.Rewrite/ModRewrite/RuleFlags.cs

```diff
@@ -0,0 +1,116 @@
+            { "co", RuleFlagType.Cookie },
+            { "cookie", RuleFlagType.Cookie },
+            { "dpi", RuleFlagType.DiscardPath },
+            { "discardpath", RuleFlagType.DiscardPath },
+            { "e", RuleFlagType.Env},
+            { "env", RuleFlagType.Env},
+            { "end", RuleFlagType.End },
+            { "f", RuleFlagType.Forbidden },
+            { "forbidden", RuleFlagType.Forbidden },
+            { "g", RuleFlagType.Gone },
+            { "gone", RuleFlagType.Gone },
+            { "h", RuleFlagType.Handler },
+            { "handler", RuleFlagType.Handler },
+            { "l", RuleFlagType.Last },
+            { "last", RuleFlagType.Last },
+            { "n", RuleFlagType.Next },
+            { "next", RuleFlagType.Next },
+            { "nc", RuleFlagType.NoCase },
+            { "nocase", RuleFlagType.NoCase },
+            { "ne", RuleFlagType.NoEscape },
+            { "noescape", RuleFlagType.NoEscape },
+            { "ns", RuleFlagType.NoSubReq },
+            { "nosubreq", RuleFlagType.NoSubReq },
+            { "p", RuleFlagType.Proxy },
+            { "proxy", RuleFlagType.Proxy },
+            { "pt", RuleFlagType.PassThrough },
+            { "passthrough", RuleFlagType.PassThrough },
+            { "qsa", RuleFlagType.QSAppend },
+            { "qsappend", RuleFlagType.QSAppend },
+            { "qsd", RuleFlagType.QSDiscard },
+            { "qsdiscard", RuleFlagType.QSDiscard },
+            { "qsl", RuleFlagType.QSLast },
+            { "qslast", RuleFlagType.QSLast },
+            { "r", RuleFlagType.Redirect },
+            { "redirect", RuleFlagType.Redirect },
+            { "s", RuleFlagType.Skip },
+            { "skip", RuleFlagType.Skip },
+            { "t", RuleFlagType.Type },
+            { "type", RuleFlagType.Type },
+            // TODO make this a load bool instead of a flag for the file and rules.
+            { "u", RuleFlagType.FullUrl },
+            { "url", RuleFlagType.FullUrl }
+            };
+
+        public IDictionary<RuleFlagType, string> FlagDictionary { get; }
+
+        public RuleFlags(IDictionary<RuleFlagType, string> flags)
+        {
+            // TODO use ref to check dictionary equality
+            FlagDictionary = flags;
+        }
+
+        public RuleFlags()
+        {
+            FlagDictionary = new Dictionary<RuleFlagType, string>();
+        }
+
+        public void AddFlag(string flag, string value)
+        {
+            RuleFlagType res;
+            if (!_ruleFlagLookup.TryGetValue(flag, out res))
+            {
+                throw new FormatException("Invalid flag");
+            }
+            AddFlag(res, value);
+        }
+        public void AddFlag(RuleFlagType flag, string value)
+        {
+            if (value == null)
+            {
+                value = string.Empty;
+            }
+            FlagDictionary[flag] = value;
+        }
+
+        public string CheckFlag(RuleFlagType flag)
```

Comment:
GetValue, or even define an indexer?
`public string this[RuleFlagType flag] { get { ...`